src/cap-description.adoc

```diff
@@ -537,6 +537,18 @@ NOTE: A capability has infinite bounds if E=CAP_MAX_E and it is not
 malformed (see xref:section_cap_malformed[xrefstyle=short]); this check is
 equivalent to _b_=0 and _t_&#8805;2^MXLEN^.
 
+[#section_cap_malformed_top_gt_inf_top]
+===== The Top Bound is Out Of Range
+
+The top bound of the <<infinite-cap>> is defined to be 2^MXLEN^.
+
+There are cases where the decoded bounds of a capability could, in theory, exceed this value.
+
+These cases are computationally expensive to detect without expanding the bounds, therefore
+all instructions such as <<CBLD>> which expand bounds and output a capability will clear
+the output tag if the top bound is out of range.
```

Comment:
I don't understand this point. No such capability is given to you in the initial system state, so of course CBLD must ensure that it doesn't give you back one of these, but that's by virtue of it never being a subset of an existing capability?

Reply:
this is like the malformed bounds case, and so I'd like (at least the option of) detagging if such a thing is observed on either input of CBLD, just like we do for malformed bounds, reserved bits set etc...

Reply:
@tariqkurd-repo : Could you provide an example situation you are thinking that CBLD would produce a capability that has this property? (i.e. its top > 2^MXLEN and is tagged).

As @jrtc27 explained, CBLD will only ever output capabilities that have equal or smaller bounds than the input capability to maintain the property that software cannot gain permissions it did not have. So, if at reset the most permissive/powerful capability provided to software is the infinite capability, then CBLD should never return a capability that is tagged and has top > infinite_cap.top.

Reply:
These are currently (mostly) undocumented malformed bounds cases, and there are many of them.
They can only be caused by fault injection or bugs e.g. the tag cache attaching the wrong tag, and so should be filtered out.
They are too expensive to detect directly from the bounds, but trivial if the bounds are expanded.
Strictly speaking nothing needs to change in the hardware as they are impossible cases, just like reserved bits set or bad permissions.
The objective is to get formal verification to work allowing _any_ bit pattern against the design, and this is one of the awkard cases, so we need to choose an approach.